tools/knb/pkg/plugin/distro.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/yaml"
+)
+
+// DistroConfig represents yaml configuration struct
+type DistroConfig struct {
+	Plugins []Plugin `yaml:"plugins"`
+}
+
+// NewDistroGenerateCmd represents plugin distro command
+func NewDistroGenerateCmd() *cobra.Command {
+	var config string
+	var generateCmd = &cobra.Command{
+		Use:   "distro",
+		Short: "Generate required files to build `kn` with inline plugins.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if _, err := os.Stat(config); os.IsNotExist(err) {
+				return fmt.Errorf("kn distro configuration file '%s' doesn't exist", config)
+			}
+			if _, err := os.Stat("cmd/kn/main.go"); os.IsNotExist(err) {
+				return fmt.Errorf("cmd/kn/main.go doesn't exist, make sure the command is executed in knative/client root directory")
+			}
+
+			fmt.Println("Generating customized kn distro:")
+			registerFile := filepath.Join("pkg", "kn", "root", "plugin_register.go")
+			if _, err := os.Stat(registerFile); !os.IsNotExist(err) {
+				fmt.Println("⚠️  plugin_register.go file already exists, trying to append imports")
+			}
+
+			rawConf, err := ioutil.ReadFile(config)
+			if err != nil {
+				return err
+			}
+			conf := &DistroConfig{}
+			if err := yaml.Unmarshal(rawConf, conf); err != nil {
+				return err
+			}
+			fmt.Println("✔  config file '" + config + "' processed")
+
+			for _, p := range conf.Plugins {
+				importPath := p.PluginImportPath
```

Comment:
Can we also simplify this for loop by having the module processing into it’s own function?

Reply:
Fixed in https://github.com/knative/client/pull/1226/commits/d940570529633941f4fc8e88d98bea95f35d3b14.